src/wayland/wl.c

```diff
@@ -635,7 +635,7 @@ static void schedule_frame_and_commit();
 
 // Draw and commit a new frame.
 static void send_frame() {
-        int scale = wl_get_scale();
+        int scale = wl_get_scale() + 0.5;
```

Comment:
This `+0.5` is doing nothing at the moment. `wl_get_scale` only returns integers now (converted to float). I would either implement a floating point scale for wayland as well (by changing a few more things to float and adding +0.5's), or changing `wl_get_scale` to return an int.

Reply:
The signature needs it to return a float because it is called as `output->get_scale()` in `draw.c`.

Reply:
Right. Still, the `+ 0.5` isn't needed here at the moment, so please remove them here where possible

Reply:
I did remove them.